source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -6,8 +6,18 @@ namespace Clusters {
 namespace Redis {
 
 bool ClusterSlot::operator==(const Envoy::Extensions::Clusters::Redis::ClusterSlot& rhs) const {
-  return start_ == rhs.start_ && end_ == rhs.end_ && primary_ == rhs.primary_ &&
-         replicas_ == rhs.replicas_;
+  if (start_ != rhs.start_ || end_ != rhs.end_ || *primary_ != *rhs.primary_ ||
+      replicas_.size() != rhs.replicas_.size()) {
+    return false;
+  }
+
+  for (auto const& replica : rhs.replicas_) {
+    if (replicas_.find(replica) == replicas_.end()) {
+      return false;
+    }
+  }
```

Comment:
Isn't this loop equivalent to just
```c++
return replicas_ == rhs.replicas_;
```

Reply:
In the test case,  "replicas_ == rhs.replicas_" returns false, but In my code, it returns true. I'm trying to find the reason for this.

Reply:
This is really not equal. Because in absl,  "operator ==" uses "==" to compare the value, and "find" uses "eq_ref" to compare the value.
![image](https://user-images.githubusercontent.com/42267882/116377778-1e07c400-a844-11eb-904b-9564419fde31.png)
![image](https://user-images.githubusercontent.com/42267882/116378486-c28a0600-a844-11eb-9551-3b9d082cba13.png)